### PR TITLE
Drop fishd check

### PR DESCRIPTION
**Pull request by zanchey**
These changes drop the old running fishd check (discussed in #3669 and #2360).

## TODOs:

- [x] Changes to fish usage are reflected in user documenation/manpages.
- [N/A] Tests have been added for regressions fixed

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,9 @@
 - `command` now accepts "-q"/"--quiet" for `command -sq $command` (#3591).
 - Selection mode (used with "begin-selection") no longer selects a character the cursor does not move over (#3684).
 - The `help` command now reads the $fish_help_browser variable (#3131).
+- fish no longer prints a warning when running instances of old versions (2.1.0
```
Comment (ridiculousfish) [inline CHANGELOG.md]: A [garden path sentence!](https://en.wikipedia.org/wiki/Garden_path_sentence)

**Comment by ridiculousfish** [comment]
Changes look good to me

File: CHANGELOG.md (hunk 1 above)
Comment (zanchey) [inline CHANGELOG.md]: Could it be better?

Comment (ridiculousfish) [inline CHANGELOG.md]: I think a small tweak to active voice is all it needs. While we're at it, I think the warning is a bit unnecessarily scary too - there's no risk of data loss for existing data.

I suggest:

"fish no longer prints a warning when it identifies a running instance of an old version (2.1.0 and earlier). Changes to universal variables may not propagate between these old versions and 2.5b1."

**Comment by zanchey** [comment]
Merged with rebase as 12ea04..4bc220f349.

**Pull request closed without merging**